core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1855,6 +1855,19 @@ object SparkContext extends Logging {
   }
 
   /**
+   * Because we can only have one active Spark Context per JVM and there are times when multiple
+   * applications may wish to share a SparkContext, this function may be used to get or instantiate
+   * a SparkContext and register it as a singleton object. 
+   */
+  private[spark] def getOrCreate(config: SparkConf = new SparkConf()): SparkContext = {
+    activeContext match {
```

Comment:
this is not your change, but can we change activeContext to an AtomicReference so it is thread safe?